### PR TITLE
Add docker containers and docker compose to the github actions flow

### PR DESCRIPTION
**Pull request by hruntov**

### CODE REVIEW

File: tests/func_test.py [hunk 1]
```diff
@@ -14,23 +14,27 @@ class BasicInstallTest(unittest.TestCase):
...
+        options.add_argument('--disable-gpu')
+        self.browser = webdriver.Remote(command_executor='http://selenium:4444/wd/hub', options=options)
+        time.sleep(5)
 
     def tearDown(self):
         self.browser.quit()
 
     def test_home_page_title(self):
-        self.browser.get("http://localhost:8000")
+        self.browser.get("http://mywebsite:8000")
```
Comment (Yevhenii-Yatchenko) [inline tests/func_test.py]: why localhost does not suite our needs?

**Comment by hruntov** [comment]
I'm using a separate docker container with selenium server for test container with the website using selenium/standalone-firefox:4.18.1-20240224 image.

File: tests/func_test.py [hunk 2]
```diff
@@ -14,23 +14,27 @@ class BasicInstallTest(unittest.TestCase):
     def setUp(self):
         options = Options()
         options.add_argument("--headless")
-        self.browser = webdriver.Chrome(options=options)
+        options.add_argument('--no-sandbox')
+        options.add_argument('--window-size=1920,1080')
+        options.add_argument('--disable-gpu')
+        self.browser = webdriver.Remote(command_executor='http://selenium:4444/wd/hub', options=options)
+        time.sleep(5)
```
Comment (Yevhenii-Yatchenko) [inline tests/func_test.py]: looks like crutch

Why shall we sleep for 5 seconds?
Can we somehow replace it with some async call?

File: tests/func_test.py [hunk 3]
```diff
@@ -142,8 +151,15 @@ def test_add_teg(self):
...
-        self.assertIn('test tag', tag_text)
+        max_attempts = 10
+        while max_attempts > 0:
+            try:
+                tag_text = self.browser.find_element(By.XPATH, '//button[@id="tag-test tag"]').text
+                self.assertIn('test tag', tag_text)
+                break
+            except StaleElementReferenceException:
+                max_attempts -= 1
+                time.sleep(1)
```
Comment (Yevhenii-Yatchenko) [inline tests/func_test.py]: ditto

File: youtube_base/youtubers/feeds.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+
+class LatestYoutubersFeed(Feed):
+    """A Feed class that generates an RSS feed for the latest Youtubers."""
+    title = 'Українська база ютуберів.'
+    desription = 'Новий ютубер у нашій базі.'
+
+    def items(self):
+        """Return the Youtuber objects that should be included in the feed."""
+        return Youtuber.objects.all()[:5]
```
Comment (Yevhenii-Yatchenko) [inline youtube_base/youtubers/feeds.py]: why do we take only five first objects?

let's invent a constant for this magic variable.

File: youtube_base/youtubers/feeds.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    def link(self, item):
+        """Return the URL for the feed itself."""
+        site = Site.objects.get_current()
+        site.domain = os.environ.get('SITE_DOMAIN')
+        site.name = os.environ.get('SITE_NAME')
+        site.save()
+
+        domain = Site.objects.get_current().domain
```
Comment (Yevhenii-Yatchenko) [inline youtube_base/youtubers/feeds.py]: 27-28 and 33-34 lines are almost equal.

Let's detach a method

def getUrl(self, url_parameters):
    domain = Site.objects.get_current().domain
    item_link = 'http://' + domain + url_parameters

File: tests/func_test.py [hunk 6]
```diff
@@ -149,17 +148,13 @@ def test_add_teg(self):
...
-            EC.presence_of_element_located((By.XPATH, '//button[@id="tag-test tag"]')))
-        max_attempts = 10
-        while max_attempts > 0:
-            try:
-                tag_text = self.browser.find_element(By.XPATH, '//button[@id="tag-test tag"]').text
-                self.assertIn('test tag', tag_text)
-                break
-            except StaleElementReferenceException:
-                max_attempts -= 1
-                time.sleep(1)
```
Comment (Yevhenii-Yatchenko) [inline tests/func_test.py]: Oh, I see

You have removed this...
Please ignore the comment in the previous commits related to these time.sleep()s

File: tests/func_test.py (hunk 2 above)
Comment (Yevhenii-Yatchenko) [reply tests/func_test.py]: please ignore

File: tests/func_test.py (hunk 3 above)
Comment (Yevhenii-Yatchenko) [reply tests/func_test.py]: please ignore

File: tests/func_test.py (hunk 1 above)
Comment (hruntov) [reply tests/func_test.py]: It's local docker variables(names of containers and thier addresses).

**Pull request merged**